EventsExpress.Core/Services/LocationService.cs

```diff
@@ -41,23 +41,35 @@ public async Task<Guid> Create(LocationDto locationDTO)
             return result.Id;
         }
 
-        public LocationDto LocationByPoint(Point point)
+        public Guid? LocationByPoint(Point point)
         {
-            var locationDTO = Mapper.Map<LocationDto>(Context.EventLocations
+            var locationDto = Mapper.Map<LocationDto>(Context.Locations
                 .Where(e =>
                     e.Point.X == point.X &&
                     e.Point.Y == point.Y)
                 .FirstOrDefault());
-            return locationDTO;
+            return locationDto.Id;
```

Comment:
let's use 
```
Context
    .Locations
    .Where(location =>
        location.Point.X == point.X &&
        location.Point.Y == point.Y)
    .Select(location => location.Id)
    .FirstOrDefault()
```

This will allow us to get rid of not-required data materialization as well as mapping

Reply:
let's skip this, we will have rid of these methods